File: hashcode2020_qualification_task/src/main/java/SolutionWriter.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.List;

public class SolutionWriter {
    public void write(Solution solution, File fileOut) throws IOException {
        try (PrintWriter writer = new PrintWriter(new FileWriter(fileOut))) {
            List<Library> libraries = solution.libraries;
            final int a = libraries.size();
            Library library;
            List<Book> books;
            int k;

            writer.println(a);
            for (int i = 0; i < a; i++) {
                library = libraries.get(i);
                books = library.booksToScan;
                k = books.size();
                if (k == 0) throw new IndexOutOfBoundsException();
                writer.print(library.id);
                writer.print(' ');
                writer.println(k);

                for (int j = 0; j < k; j++) {
                    Book book = books.get(j);
                    if (j > 0) writer.print(' ');
                    writer.print(book.id);
                }
                writer.println();
            }
            writer.flush();
        }
    }
}

File: pizza.2021/src/main/java/SolutionWriter.kt
import java.io.File
import kotlin.Throws
import java.io.IOException
import java.io.PrintWriter
import java.io.FileWriter

class SolutionWriter {
    @Throws(IOException::class)
    fun write(solution: Solution, fileOut: File) {
        PrintWriter(FileWriter(fileOut)).use { writer ->
            val teams = solution.teams

            writer.print(teams.size)
            writer.print(EOL)
            for (team in teams) {
                writer.print(team.numberOfMembers)
                for (p in team.pizzas) {
                    writer.print(' ')
                    writer.print(p.index)
                }
                writer.print(EOL)
            }
            writer.flush()
        }
    }

    companion object {
        private const val EOL = '\n'
    }
}